### ISSUE TITLE
Update documentation on freedesktop.org

### ISSUE THREAD

**Issue by debug-richard** [PackageKit/PackageKit#577]
I have implemented the PackageKit based on the documentation from https://www.freedesktop.org/software/PackageKit/gtk-doc/ .
Unfortunately, the documentation was last updated around 2017/2018 and there is a lot of missing or outdated information.
In particular, the PolicyKit permissions and the new enums/constants are missing/wrong.

**Comment by hughsie** [comment]
FWIW, the process used to be:

* build fwupd
* run docs/upload.sh

But I don't think the gtk-docs are generating now for some reason.

**Comment by ximion** [comment]
Yes, I actually had the same issue in AppStream and implemented a workaround there. We can probably do the same here, I can look into that :-)

**Comment by sidt4** [comment]
I think we need some effort to refresh the content under 
https://www.freedesktop.org/software/PackageKit

E.g. https://www.freedesktop.org/software/PackageKit/pk-users.html is very old ( missing GNOME Software , mentions Anjutha etc )

**Comment by sidt4** [comment]
In `gnome-software` , we wanted to print the running `PackageKit` version in the logs (as it was currently printing `PackageKit` version it was built with, which was not always correct). This was implemented and merged in https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/1904

Initially in the MR above, I took the D-bus code route (as below), as my search in `Gtk-doc` for `pk version` search string didn't offer anything meaningful, and I saw the version properties available through `d-feet` / `D-Spy` app.

E.g.

```c
static void
gs_plugin_packagekit_version_check_async (GsPlugin            *plugin,
					  GCancellable        *cancellable)
{
	g_dbus_connection_call (gs_plugin_get_system_bus_connection (plugin),
				"org.freedesktop.PackageKit",
				"/org/freedesktop/PackageKit",
				"org.freedesktop.DBus.Properties",
				"GetAll",
				g_variant_new ("(s)", "org.freedesktop.PackageKit"),
				G_VARIANT_TYPE ("(a{sv})"),
				G_DBUS_CALL_FLAGS_NONE,
				-1,
				cancellable,
				gs_plugin_packagekit_version_check_async_cb,
				NULL);
}
```

But, later I switched to using `PkControl` after suggestion from Milan.  Looking into the `pk-control.c`, we do have properties for the version (`version-major`, `version-minor`, `version-micro` etc), but they were not listed in Gtk-doc search for `"pk version"` string. I spent some time trying to figure why the properties were not added to `PackageKit-PkControl.html`, but didn't progress much on that.

Any clues ?
